src/app/order/order.component.ts

```diff
@@ -57,35 +71,42 @@ export class OrderComponent implements OnInit, OnDestroy {
   }
 
   createOrderForm() {
-    const restaurantId = this.restaurant && this.restaurant._id || null;
-    this.orderForm = this.formBuilder.group({
+    const restaurantId = this.restaurant && this.restaurant._id;
+    this.orderForm = this.formBuilder.nonNullable.group({
       restaurant: [restaurantId],
-      name: [null, Validators.required],
-      address:  [null, Validators.required],
-      phone: [null,  Validators.required],
-      items: [[], minLengthArray(1)]
+      name: ['', Validators.required],
+      address: ['', Validators.required],
+      phone: ['', Validators.required],
+      items: [[] as Item[], minLengthArray(1)],
     });
     this.onChanges();
   }
 
   onChanges() {
-    this.orderForm.get('items')?.valueChanges.pipe(takeUntil(this.unSubscribe)).subscribe((val: Item[]) => {
-      this.orderTotal = this.itemTotal.transform(val);
-    });
+    this.orderForm
+      ?.get('items')
+      ?.valueChanges.pipe(takeUntil(this.unSubscribe))
```

Comment:
non-blocker, but should we use the built-in https://angular.io/api/core/rxjs-interop/takeUntilDestroyed operator?

Reply:
I am in favor of updating it. There might be other similar things we can update too.

Reply:
@tehfedaykin what do you think?